### PR TITLE
Conflict check

### PR DESCRIPTION
**Pull request by KevinDeMars**

### CODE REVIEW

File: food-truck-api/src/main/java/food/truck/api/routes/RouteLocation.java [hunk 1]
```diff
@@ -36,4 +37,22 @@
     @Convert(converter = PositionConverter.class)
     @Column(nullable = false)
     Position position;
+
+    @JsonIgnore
+    public boolean wrapsAroundMidnight() {
+        return exitTime.isBefore(arrivalTime);
+    }
+
+    // I have no idea how I came up with this but the tests seem to work
```
Comment (MicahSchiewe) [inline food-truck-api/src/main/java/food/truck/api/routes/RouteLocation.java:46]: Story of my coding life: I have 0 idea why it works, but it does :P.

File: food-truck-api/src/main/java/food/truck/api/routes/RouteLocation.java [hunk 2]
```diff
@@ -36,4 +37,22 @@
...
+    // 4 cases, depending on which locations wrap around midnight (UTC)
+    public boolean timeConflictsWith(RouteLocation other) {
+        if (!wrapsAroundMidnight() && !other.wrapsAroundMidnight())
+            return !arrivalTime.isAfter(other.exitTime) && !exitTime.isBefore(other.arrivalTime);
+        else if (wrapsAroundMidnight() && !other.wrapsAroundMidnight())
```
Comment (MicahSchiewe) [inline food-truck-api/src/main/java/food/truck/api/routes/RouteLocation.java:51]: The code on these two 'one wraps around at midnight' branches are identical, just flipped around the `||`.

File: food-truck-api/src/main/java/food/truck/api/truck/TruckService.java [hunk 3]
```diff
@@ -95,8 +91,11 @@ public Truck updateTruck(
                     .filter(tag -> !tag.isEmpty())
                     .collect(Collectors.toSet());
             truck.setTags(tags2);
+        } else {
+            truck.setTags(null);
```
Comment (MicahSchiewe) [inline food-truck-api/src/main/java/food/truck/api/truck/TruckService.java:95]: I'll need to check whether I assumed non-NULL truck tags while performing the semantic analysis stuff. There's a chance I did.

**Review by MicahSchiewe — APPROVED** [review]
LGTM

File: food-truck-api/src/main/java/food/truck/api/truck/TruckService.java [hunk 4]
```diff
@@ -49,17 +49,13 @@
...
-
     public Truck createTruck(Long userId, String name) {
         var t = new Truck();
         t.setName(name);
         t.setUserId(userId);
         t.setDescription("");
         t.setTags(new HashSet<>());
-        return saveTruck(t);
+        return truckRepository.save(t);
     }
```
Comment (nnazo) [inline food-truck-api/src/main/java/food/truck/api/truck/TruckService.java:59]: Are we getting rid of the `saveTruck` method? I'll need to update that in my PR as well if so.

**Review by nnazo — APPROVED** [review]
Looks good to go.

**Pull request merged**